File: advent_of_code/2023/day14/part1.py
from sys import stdin

arr = []
res = 0

def Reverse(tup):
    new_tup = tup[::-1]
    return new_tup

def press(arr):
    res = 0
    arr = list(zip(*arr[::-1]))

    for i in range(len(arr)):
        arr[i] = Reverse(arr[i])

    for line in arr:
        l = len(line)
        l2 = l
        for c in line:
            if c == 'O':
                res += l
                l-=1
            elif c == '#':
                l = l2-1
            l2 -= 1
    return res

for line in stdin:
    line = line[:-1]
    arr.append(line)

print(press(arr))

File: advent_of_code/2024/day04/part1.cpp
#include <iostream>

using namespace std;

int main(){
    int n = 146;
    string arr[150];
    for(int i = 0; i < n; i++){
        cin >> arr[i];
    }

    int res = 0;
    
    for(int i = 3; i < 145; i ++){
        for(int k = 3; k < 145; k++){
            if(arr[i][k] == 'X'){
                if(arr[i][k+1] == 'M' and arr[i][k+2] == 'A' and arr[i][k+3] == 'S')
                    res++;
                if(arr[i][k-1] == 'M' and arr[i][k-2] == 'A' and arr[i][k-3] == 'S')
                    res++;
                if(arr[i+1][k] == 'M' and arr[i+2][k] == 'A' and arr[i+3][k] == 'S')
                    res++;
                if(arr[i-1][k] == 'M' and arr[i-2][k] == 'A' and arr[i-3][k] == 'S')
                    res++;

                if(arr[i-1][k-1] == 'M' and arr[i-2][k-2] == 'A' and arr[i-3][k-3] == 'S')
                    res++;
                if(arr[i-1][k+1] == 'M' and arr[i-2][k+2] == 'A' and arr[i-3][k+3] == 'S')
                    res++;
                if(arr[i+1][k-1] == 'M' and arr[i+2][k-2] == 'A' and arr[i+3][k-3] == 'S')
                    res++;
                if(arr[i+1][k+1] == 'M' and arr[i+2][k+2] == 'A' and arr[i+3][k+3] == 'S')
                    res++;
            }
        }
    }

    cout << res;

    return 0;
}
